build_daemon/lib/client.dart

```diff
@@ -123,6 +121,8 @@ class BuildDaemonClient {
   }
 }
 
+class MissingPortFileException extends Error {}
```

Comment:
why Error and not Exception?

Come to think of it, OptionsSkew and VersionSkew should probably be Exception as well. They are things that may happen outside of the programmer's control and don't indicate a bug.